src/wallet/wallet.cpp

```diff
@@ -1572,38 +1572,49 @@ void CWallet::SyncTransaction(const CTransaction& tx, const CBlock* pblock)
     MarkAffectedTransactionsDirty(tx);
 }
 
-void CWallet::SyncCertificate(const CScCertificate& cert, const CBlock* pblock)
+void CWallet::SyncCertificate(const CScCertificate& cert, const CBlock* pblock, int bwtMaturityHeight)
 {
     LOCK2(cs_main, cs_wallet);
     if (!AddToWalletIfInvolvingMe(cert, pblock, true))
         return; // Not one of ours
 
     MarkAffectedTransactionsDirty(cert);
+    std::map<uint256, std::shared_ptr<CWalletTransactionBase>>::iterator itCert = mapWallet.find(cert.GetHash());
+    if (itCert != mapWallet.end())
```

Comment:
If cert is not found that should be marked as an error

Reply:
I think it should never happen, and I should probably assert cert presence and nature (i.e. IsCertificate) or throw. As of now, I'd go for the first

Reply:
Your comment and my previous one are wrong.
The reason is that a certificate will appear in a mapWallet of a specific node only if it is originated from a directed to one of the addresses in the wallet.
There is nothing to be done here if certificate is not found. Simply return